app/models/user.rb

```diff
@@ -28,7 +28,7 @@ def self.find_or_create_from_provider_data(provider_data, feature_flag_id:)
   end
 
   def self.find_or_create_from_tra_data_on_uid(provider_data, feature_flag_id:)
-    user_from_provider_data = find_or_initialize_by(provider: provider_data.provider, uid: provider_data.uid)
+    user_from_provider_data = find_or_initialize_by(email: provider_data.info.email)
```

Comment:
@slawosz if we don't find the user by the `uid`, if the user changes the email on GAI, then we won't be able to find it.

Reply:
I wasn't probably very clear :)

When a user is authenticated in GAI, on NPQ we should only care about the UID from that user in GAI and update all the other information. This information should then be also passed to ECF (if changes)